src/log.py

```diff
@@ -570,26 +570,108 @@ def format(self, record):
         return msg
 
     def identify_data_to_redact(self, record):
+        # This matches stuff like db_pwd=the_secret or admin_password=other_secret
+        # (the secret part being at least 3 chars to avoid catching some lines like just "db_pwd=")
+        # Some names like "key" or "manifest_key" are ignored, used in helpers like ynh_app_setting_set or ynh_read_manifest
+        secret_keys = 'pass|secret|token|salt'
+        secret_keys = f"{secret_keys}|{secret_keys.upper()}|pwd"
+        # Avoid to match some non relevant var compound with 'key'
+        secret_keys_regex = f"\w*({secret_keys})\w*|local key_?| key\w*|\w+key\w*|\w*KEY\w+|\w+KEY"
+        operator_regex = r'(\s+\-\-value)?='
+        value_regex = r"'([^']|'\''){5,}'|-----BEGIN.*-----END|\S{5,}"
+        redact_regex = re.compile(f"({secret_keys_regex}){operator_regex}({value_regex})")
```

Comment:
I think it would be better to match the value inside `-----BEGIN.*-----` and `-----END.*-----`, because then we know that the value was a certificate or a private key.

Also, we should not redact the value with quotes, because sometimes (in bash conditions for instance) the quotes aren't there anymore, so we would miss the value to redact.

So I would suggest something like that:
```suggestion
        value_regex = r"([^']|'\''){3,}?|\S{3,}?"
        redact_regex = re.compile(rf"({secret_keys_regex}){operator_regex}'?(-----BEGIN.+?-----)?({value_regex})(-----END.+?-----)?'?$")
```
(I didn't understant which case you want to cover with `'([^']|'\''){5,}'` in the value regex ?

Reply:
Bash set -x display `'` as `'\''`
```
$ set -x
$ echo "L'arbre"
+ echo 'L'\''arbre'
L'arbre
```